engine/worker/internal/run.go

```diff
@@ -126,7 +126,13 @@ func (w *CurrentWorker) runJob(ctx context.Context, a *sdk.Action, jobID int64,
 		// Reset step log line to 0
 		w.stepLogLine = 0
 
-		ctx = workerruntime.SetStepOrder(ctx, jobStepIndex)
+		ctx = workerruntime.SetStepOrder(ctx, jobStepIndex+1)
```

Comment:
why +1 ?

Reply:
StepIndex is optional is pi_ref struct . (because hatchery does not provide it), so json tag has omitempty.
With value 0: field is not present. So step starts at 1